crates/forge/src/running.rs

```diff
@@ -67,18 +83,54 @@ pub(crate) fn run_from_test_case(
         footer.iter()
     );
     let (hints_dict, string_to_hint) = build_hints_dict(instructions.clone());
-    let core_cairo_hint_processor = CoreCairoHintProcessor {
-        runner: Some(runner),
-        starknet_state: StarknetState::default(),
-        string_to_hint,
-        run_resources: RunResources::default(),
+
+    let block_context = build_block_context();
```

Comment:
Is this and whole logic below copied from somewhere?

Reply:
nah

Reply:
Losely inspired by https://github.com/foundry-rs/starknet-foundry/blob/7a46ea32d5bba4d8f4dc9928b5ff6216a73de946/crates/cheatnet/src/execution/cairo1_execution.rs

but different enough that I haven't considered linking it

Reply:
You think we should?

Reply:
I think it could be useful in case we need to change something. To look for inspiration I guess 😅

Reply:
DONE